lib/src/presentation/bloc/character_bloc.dart

```diff
@@ -32,9 +34,18 @@ class CharacterBloc implements BlocInterface {
   }
 
   @override
-  void fetchAllCharacters() async {
-    _generalCharacter = await _characterUseCase.fetchAllCharacters();
-    _nextPageCharacter = _generalCharacter.info.next!;
+  void fetchFirstCharacters() async {
+    final _generalCharacterFirst = await _characterUseCase.fetchAllCharacters();
+    _nextPageCharacter = _generalCharacterFirst.info.next!;
+    if (_generalCharacterFirst.results.isNotEmpty) {
+      await _characterDatabase.dropDatabase();
+      await _characterDatabase
+          .addCharacterToDataBase(_generalCharacterFirst.results);
+    }
+    _generalCharacter = GeneralCharacter(
+      info: _generalCharacterFirst.info,
+      results: await _characterDatabase.fetchCharacterFromDataBase(),
+    );
```

Comment:
This could be a better approach if you inside the initialization you can listen from changes of the stream and then you assign the value that you listen into _generalCharacter and manage the logic to do the addAll from the list of characters or the replace of next page url

Reply:
done